sklearn/ensemble/_forest.py

```diff
@@ -511,8 +512,10 @@ def _compute_oob_predictions(self, X, y):
         oob_pred : ndarray of shape (n_samples, n_classes, n_outputs) or \
                 (n_samples, 1, n_outputs)
             The OOB predictions.
-      """
-        X = self._validate_data(X, dtype=DTYPE, accept_sparse="csr", reset=False)
+        """
+        # Prediction requires X to be in CSR format
+        if issparse(X):
+            X = check_array(X, accept_sparse="csr", force_all_finite=True)
```

Comment:
Why `force_all_finite=True` here if input validation has already been performed in the caller?

Wouldn't the following be enough?

```suggestion
            X = X.tocsr()
```